examples/md/md_with_nf.py

```diff
@@ -0,0 +1,253 @@
+    MD.optimize_set_up(m)
+
+    # I noticed that area will hit the upper bound of 150m set in the flowsheet. Removing upper bounds.
+    m.fs.MD.area.setub(None)
+    m.fs.MD.length.setub(None)
+    m.fs.MD.width.setub(None)
+
+    # For safe measure--invoking interval initializer before solving
+    MD.interval_initializer(m)
+    MD.solve(m)
+    m.fs.MD.report()
+
+    # assert False
+    # Notably, system-level recovery rate of water is set to 50%. I think the 5 kg/s case couldn't solve because more membrane area would be needed to achieve 50% recovery.
+    m.fs.overall_recovery.display()
+
+    # NF Permeate conditions
+    m.fs.feed.flow_mass_phase_comp[0,"Liq","H2O"].fix(7.2258)
+    m.fs.feed.flow_mass_phase_comp[0,"Liq","TDS"].fix(0.64491)
+    res = MD.solve(m, tee=False)
+    if check_optimal_termination(res):
+        # m.fs.MD.report()
+        m.fs.MD.area.display()
+    else:
+        print("SOLVE FAILED")
+        infeas.print_infeasible_constraints(m)
+    
+    
+       #%% bring in prommis costing
+    from prommis.uky.costing.ree_plant_capcost import QGESSCosting, QGESSCostingData
+    # Add NF for costing only?
+    import watertap.property_models.multicomp_aq_sol_prop_pack as props
+
+    from watertap.unit_models.nanofiltration_ZO import NanofiltrationZO
+    from watertap.unit_models.pressure_changer import Pump  
+    import json
+
+    with open(r"C:\Users\Adam\my-nawi-hub\SA2_009_004_EY24\solute_parameters.json") as f:
+        solute_data = json.load(f)
+
+    # solute list
+    solute_list = list(solute_data.keys())
+    mw_data = {key: solute_data[key]['mw'] for key in solute_list}
+    charge = {key: solute_data[key]['charge'] for key in solute_list}
+    diffusivity = {("Liq",key): 1e-9 for key in solute_list}
+
+    m.fs.properties = props.MCASParameterBlock(solute_list=solute_list,
+                                               mw_data=mw_data,
+                                               charge=charge,
+                                               diffusivity_data=diffusivity,
+                                               density_calculation=props.DensityCalculation.seawater,
+                                               material_flow_basis=props.MaterialFlowBasis.mass)
+
+    # create units
+    # m.fs.feed = Feed(property_package=m.fs.properties)
+    # m.fs.product = Product(property_package=m.fs.properties)
+    # m.fs.disposal = Product(property_package=m.fs.properties)
+    m.fs.nf = NanofiltrationZO(property_package=m.fs.properties)
+    m.fs.P1 = Pump(property_package=m.fs.properties)
+
+    m.fs.P1.efficiency_pump.fix(0.80)  # pump efficiency [-]
+    m.fs.P1.outlet.pressure[0].fix(10e5)
+
+    # fully specify system
+    m.fs.nf.properties_permeate[0].pressure.fix(101325)
+    m.fs.nf.recovery_vol_phase.fix(0.5)
+    m.fs.nf.flux_vol_solvent.fix(1.446759259259259e-5)
+    m.fs.nf.area.fix(499.44685)
```

Comment:
injecting nf and nf pump merely for costing. I specified the nf membrane area that would result from running NF separately. Double check all specs match what we expect so that costing is not misrepresented for NF pretreatment train